### DISCUSSION TITLE
Pros and cons having Master and Volume on same machine

### DISCUSSION THREAD

**Discussion by shorty666** [seaweedfs/seaweedfs#1934] [Q&A] [answered]
Can someone explain the pros and cons of running the master and volume on the same server vs. each on seperate machines?

In particular I'm interested in things like performance, maintenance and failure tolerance.

Thanks in advance!

**Comment by chrislusf** [comment, marked as answer]
Separated master and volume servers is recommended, good for performance, maintenance and failure tolerance.

Cons is cost.

**Reply by shorty666** [reply]
OK, thank you for your answer.
